Classes/Infrastructure/DeepL/DeepLTranslationService.php

```diff
@@ -76,191 +57,106 @@ class DeepLTranslationService implements TranslationServiceInterface
-                $part = IgnoredTermsUtility::wrapIgnoredTerms($part, $this->settings['ignoredTerms']);
-            }
-
-            $body .= '&text=' . urlencode($part);
+        // wrap ignoredTerms
+        if (isset($this->settings['ignoredTerms']) && count($this->settings['ignoredTerms']) > 0) {
+            $valuesWithMaskedTerms = array_map(
+                fn(string $text) => IgnoredTermsUtility::wrapIgnoredTerms($text, $this->settings['ignoredTerms']),
+                $values
+            );
+        } else {
+            $valuesWithMaskedTerms = $values;
         }
 
-        $apiRequest = $this->createRequest('translate', 'POST', $body);
-
-        $browser = $this->getBrowser();
-
-        $attempt = 0;
-        $maximumAttempts = $this->settings['numberOfAttempts'];
-        $apiResponse = null;
-        do {
-            $attempt++;
-            try {
-                $apiResponse = $browser->sendRequest($apiRequest);
-                break;
-            } catch (CurlEngineException $e) {
-                if ($attempt === $maximumAttempts) {
-                    return $texts;
-                }
+        try {
+            /**
+             * @var TextResult[]|TextResult $results
+             */
+            $results = $client->translateText(
+                $valuesWithMaskedTerms,
+                $sourceLanguage,
+                $targetLanguage,
+                $translateTextOptions
+            );
 
-                sleep(1);
-                continue;
+            if ($results instanceof TextResult) {
+                $results = [$results];
             }
-        } while ($attempt <= $maximumAttempts);
 
-        if (is_null($apiResponse)) {
-            return $texts;
-        } elseif ($apiResponse->getStatusCode() == 200) {
-            $returnedData = json_decode($apiResponse->getBody()->getContents(), true);
-            if (is_null($returnedData)) {
-                return array_replace($texts, $cachedEntries);
-            }
             $translations = array_map(
-                function ($part) {
-                    return IgnoredTermsUtility::unwrapIgnoredTerms($part['text']);
-                },
-                $returnedData['translations']
+                fn (TextResult $textResult) => IgnoredTermsUtility::unwrapIgnoredTerms($textResult->text),
+                $results
             );
 
             $translationWithOriginalIndex = array_combine($keys, $translations);
 
-            if ($isCacheEnabled) {
+            if ($this->translationCache?->isEnabled()) {
                 foreach ($translationWithOriginalIndex as $i => $translatedString) {
                     $originalString = $texts[$i];
-                    $this->translationCache->set(self::getEntryIdentifier($originalString, $targetLanguage, $sourceLanguage), $translatedString);
+                    $this->translationCache->set($originalString, $translatedString, $sourceLanguage, $targetLanguage);
                 }
             }
```

Comment:
@mficzel Is there a retry concept in the DeepL api? Because if not I would argue to keep it.